Source/WebKit/UIProcess/WebBackForwardList.cpp

```diff
@@ -601,21 +603,21 @@ static WebBackForwardListItem* itemSkippingBackForwardItemsAddedByJSWithoutUserG
     } else {
         // If going forward and there are items that we created by JS without user interaction, move forward to the last
         // one in the series.
-        auto* nextItem = backForwardList.itemAtIndex(itemIndex + 1);
+        RefPtr nextItem = backForwardList.itemAtIndex(itemIndex + 1);
         while (nextItem && nextItem->wasCreatedByJSWithoutUserInteraction()) {
-            item = nextItem;
+            item = WTFMove(nextItem);
             nextItem = backForwardList.itemAtIndex(++itemIndex);
```

Comment:
```suggestion
            item = std::exchange(nextItem, backForwardList.itemAtIndex(++itemIndex));
```
I like how std::exchange avoids the window between two lines of code where `nextItem` has been "moved out of".